Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol CacheDirectoriesProviding: AnyObject {
+    /// Returns the cache directory
+    var cacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the build artifacts are cached.
+    var buildCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where hashes of modules that have been a part of successful test are cached
+    var testsCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the projects generated for automation tasks are generated to
+    var projectsCacheDirectory: AbsolutePath { get }
```

Comment:
Nitpick: `generatedAutomationProjectsDirectory`. It's longer, but more explicit about what the directory is for.

Reply:
This was all existing naming, yet I am happy to change it :)